src/DB.hs

```diff
@@ -77,3 +101,39 @@ stubbedInitialDataMap = Map.fromList
 stubbedBlacklistedPools :: [PoolHash]
 stubbedBlacklistedPools = []
 
+postgresqlDataLayer :: DataLayer
+postgresqlDataLayer = DataLayer
+    { dlGetPoolMetadata     = \poolHash -> do
+        txMetadata' <- runDbAction Nothing $ queryTxMetadata (encodeUtf8 $ getPoolHash poolHash)
+
+        let txMetadata = either (\_ -> panic "EROR!") (\m -> m) txMetadata'
+
+        let metadata :: Text
+            metadata = txMetadataMetadata txMetadata
+
+        --BS.putStrLn metadata
+        --putTextLn $ decodeUtf8 metadata
+
+        --return $ first (\m -> UnknownError (toS m)) $ eitherDecode $ BL.fromStrict metadata
+        return $ first (\m -> UnknownError (toS m)) $ eitherDecode $ BL.fromStrict (encodeUtf8 metadata)
+
+    , dlGetPoolMetadataSimple = \poolHash -> do
+        txMetadata <- runDbAction Nothing $ queryTxMetadata (encodeUtf8 $ getPoolHash poolHash)
+        return (txMetadataMetadata <$> txMetadata)
+
+    , dlAddPoolMetadata     = \poolHash poolMetadata -> panic "To implement!"
+
+    , dlAddPoolMetadataSimple     = \poolHash poolMetadata -> do
+        let poolHashBytestring = (encodeUtf8 $ getPoolHash poolHash)
+        let poolEncodedMetadata = poolMetadata
+        let hashFromMetadata = B16.encode $ Crypto.digest (Proxy :: Proxy Crypto.Blake2b_256) $ (encodeUtf8 poolEncodedMetadata)
+
+        when (hashFromMetadata /= poolHashBytestring) $
+            panic "TxMetadataHashMismatch"
```

Comment:
I get from this line that `poolHash` is actually not the pool id but the hash of the metadata? This feels rather confusing with the name "poolHash"?

Reply:
A lot of things in this PR are confusing, will address that.